include/bout/index_derivs.hxx

```diff
@@ -308,7 +308,7 @@ DEFINE_UPWIND_DERIV(VDDX_WENO3, "W3", 2, DERIV::Upwind) { // No vec
     r = (WENO_SMALL + SQ(f.pp - 2.0 * f.p + f.c))
         / (WENO_SMALL + SQ(f.p - 2.0 * f.c + f.m));
 
-    deriv = (-f.m + 3. * f.c - 3. * f.p + f.pp);
+    deriv = -(-f.m + 3. * f.c - 3. * f.p + f.pp);
```

Comment:
This doesn't look right to me. In the limit of smooth f, r->1. Then the left stencil gives `1./6.*f.mm - f.m + 1./2.*f.c + 1./3.*f.p`, which is a 3rd order first derivative (see http://web.media.mit.edu/~crtaylor/calculator.html with points at -2,-1,0,1). Before this change the right stencil is similar, but after it the right stencil would give `-2./3.*f.m + 1./2.*f.c + 1./6.*f.pp`, which isn't a 3rd order first derivative. Maybe there was a typo in the paper you cited @d7919?

Reply:
Yes that's definitely possible.

Reply:
(in the other limit both branches reduce to a second order central difference, I'm a bit surprised it's second order rather than first order and one sided but that's without thinking about it too hard)